### PR TITLE
Upgrade to k8s API v0.24.0

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

### CODE REVIEW

File: pkg/server/server_test.go [hunk 1]
```diff
@@ -442,7 +442,7 @@ func TestAuthenticateVerifierRoleMapping(t *testing.T) {
 	}
 	h := setup(&testVerifier{err: nil, identity: identity})
 	h.mappers = []mapper.Mapper{file.NewFileMapperWithMaps(map[string]config.RoleMapping{
-		"arn:aws:iam::0123456789012:role/test": config.RoleMapping{
+		"arn:aws:iam::0123456789012:role/test": {
```
Comment (jaypipes) [inline pkg/server/server_test.go:445]: any reason for the changes introduced in this file?

**Review by jaypipes — COMMENTED** [review]
I'm OK with the upgrade to k8s.io API machinery v0.24.0 but am wondering why there are unrelated changes included in this PR for test files.

**Review by jaypipes — APPROVED** [review]
@ahmedwaleedmalik thanks for providing me some insights into those unrelated test changes. This is fine by me. I will let @nckturner do the final merge.

**Comment by ahmedwaleedmalik** [comment]
So the tests are failing but I can't even run them locally?

**Comment by tamalsaha** [comment]
any updates on this? Thanks.

**Pull request closed without merging**